test/invariant/actors/Proposer.t.sol

```diff
@@ -0,0 +1,130 @@
+pragma solidity ^0.8.19;
+
+import { BaseActor, IERC20, IERC20Metadata, AngleGovernor, TestStorage } from "./BaseActor.t.sol";
+import { console } from "forge-std/console.sol";
+import { IGovernor } from "oz/governance/IGovernor.sol";
+import { ProposalStore, Proposal } from "../stores/ProposalStore.sol";
+import { IERC5805 } from "oz/interfaces/IERC5805.sol";
+import { TimestampStore } from "../stores/TimestampStore.sol";
+
+contract Proposer is BaseActor {
+    AngleGovernor internal _angleGovernor;
+    ProposalStore public proposalStore;
+    IERC5805 public veANGLEDelegation;
+    TimestampStore public timestampStore;
+
+    constructor(
+        AngleGovernor angleGovernor,
+        IERC20 _agToken,
+        uint256 nbrVoter,
+        ProposalStore _proposalStore,
+        IERC5805 _veANGLEDelegation,
+        TimestampStore _timestampStore
+    ) BaseActor(nbrVoter, "Proposer", _agToken) {
+        _angleGovernor = angleGovernor;
+        proposalStore = _proposalStore;
+        veANGLEDelegation = _veANGLEDelegation;
+        timestampStore = _timestampStore;
+    }
+
+    function propose(uint256 value) public useActor(1) {
+        address[] memory targets = new address[](1);
+        bytes[] memory datas = new bytes[](1);
+        uint256[] memory values = new uint256[](1);
+        string memory description = "Test Proposal";
+        targets[0] = address(_angleGovernor);
+        datas[0] = "";
+        values[0] = value;
+
+        uint256 proposalId = _angleGovernor.hashProposal(targets, values, datas, keccak256(bytes(description)));
+        if (_angleGovernor.proposalSnapshot(proposalId) != 0) {
+            vm.expectRevert(
+                abi.encodeWithSelector(
+                    IGovernor.GovernorUnexpectedProposalState.selector,
+                    proposalId,
+                    _angleGovernor.state(proposalId),
+                    bytes32(0)
+                )
+            );
+        }
+        _angleGovernor.propose(targets, values, datas, description);
+
+        // Add to the store
+        proposalStore.addProposal(targets, values, datas, keccak256(bytes(description)));
+    }
+
+    function execute(uint256 proposalId) public useActor(1) {
+        if (proposalStore.nbProposals() == 0) {
+            return;
+        }
+        Proposal memory proposal = proposalStore.getRandomProposal(proposalId);
+        uint256 proposalHash = _angleGovernor.hashProposal(
+            proposal.target,
+            proposal.value,
+            proposal.data,
+            proposal.description
+        );
+        uint256 proposalSnapshot = _angleGovernor.proposalSnapshot(proposalHash);
+        timestampStore.increaseCurrentTimestamp(_angleGovernor.proposalDeadline(proposalHash));
+        timestampStore.increaseCurrentBlockNumber(
+            _angleGovernor.$snapshotTimestampToSnapshotBlockNumber(proposalSnapshot)
+        );
+        vm.warp(_angleGovernor.proposalDeadline(proposalHash) + 1);
```

Comment:
why not using the timestamp given by `timestampStore`?

Reply:
to be sure that if the execute is ready, it will execute

Reply:
I have another function tryToExecute will just try to execute without any guarantee and this one is supposed to be more successfull

Reply:
To clarify I misunderstood the question and made the change right away